localstack-core/localstack/services/events/utils.py

```diff
@@ -293,3 +295,29 @@ def is_nested_in_string(template: str, match: re.Match[str]) -> bool:
         return False
 
     return left_quote != -1
+
+
+def create_segment_from_trace_header(trace_header: TraceHeader) -> None:
+    segment = xray_recorder.begin_segment(name="events.put_events")
+    segment.trace_id = trace_header.root
+    segment.parent_id = trace_header.parent
+    if trace_header.sampled == "1":
+        segment.sampled = True
+    elif trace_header.sampled == "0":
+        segment.sampled = False
+
+
+def get_trace_header_str_from_segment():
+    # Get the current segment
+    segment = xray_recorder.current_segment()
```

Comment:
question: what if we import the `xray_recorder` in the code somewhere else and start using it? wouldn't this start failing weirdly? as `xray_recorder` is a global variable?